executor/src/witgen/global_constraints.rs

```diff
@@ -214,10 +214,8 @@ fn propagate_constraints<T: FieldElement>(
     let mut remove = false;
     match identity.kind {
         IdentityKind::Polynomial => {
-            if let Some(p) = is_binary_constraint(identity.expression_for_poly_id()) {
-                assert!(known_constraints
-                    .insert(p, RangeConstraint::from_max_bit(0))
-                    .is_none());
+            if let Some((p, constraint)) = is_binary_constraint(identity.expression_for_poly_id()) {
```

Comment:
Now to use the root-finding function in the most general case (that includes the binary one), you should do the following:

Set the constraint to from_range(min, max) and set `remove` to true if all values from min to max occur in the list.

Reply:
I didn't get the it well, can you give more details? You mean I don't use `from_mask` and use `from_range` instead? and don't try to make a binary mask which covers all numbers?

Reply:
exactly